src/main/java/org/jenkinsci/plugins/codesonar/services/HubVersionService.java

```
@@ -0,0 +1,181 @@
+     * @param clientName the client name, in this case "jenkins"
+     * @param clientVersion the client version
+     * @throws IOException thrown when the communication with the hub fails
+     */
+    private VersionCompatibilityInfo fetchVersionCompatibilityInfo(URI baseHubUri, String clientName, String clientVersion) {
+        
+        URI resolvedURI = baseHubUri;
+        
+        resolvedURI = baseHubUri.resolve(String.format("/command/check_version/%s/?version=%s&capability=openapi", clientName, clientVersion));
+        LOGGER.log(Level.INFO, "Calling " + resolvedURI.toString());
+        
+        HttpResponse resp;
+        try {
+            resp = httpService.execute(Request.Get(resolvedURI))
+                    .returnResponse();
+        } catch (IOException e) {
+        	LOGGER.log(Level.WARNING, String.format("[CodeSonar] failed to get a response. %n[CodeSonar] IOException: %s%n[CodeSonar] Stack Trace: %s", e.getMessage(), Throwables.getStackTraceAsString(e)));
+            return null;
+        }
+        
+        if(resp.getStatusLine() == null) {
+        	LOGGER.log(Level.INFO, String.format("[CodeSonar] not able to read http status."));
+            return null;
+        }
+        
+        if(resp.getStatusLine().getStatusCode() == 404) {
+            //Hub might respond with an HTTP 404, we want to keep track this special case
+        	LOGGER.log(Level.INFO, String.format("[CodeSonar] specified endpoint seems not to exist on the hub. %n[CodeSonar] response is \"%d, %s\"", resp.getStatusLine().getStatusCode() , resp.getStatusLine().getReasonPhrase()));
+            return null;
+        } else if(resp.getStatusLine().getStatusCode() != 200) {
+            //Hub returned an unexpected response
+        	LOGGER.log(Level.INFO, String.format("[CodeSonar] response is not successfull. %n[CodeSonar] response is \"%d, %s\"", resp.getStatusLine().getStatusCode() , resp.getStatusLine().getReasonPhrase()));
+            return null;
+        }
```

If we don't get a 200 or a 404, then we need to log the response body in addition to the status and reason phrase.

Also, I think we should raise an exception instead of returning `null` since this is an exceptional case.  Imagine if you get a 500 from a 7.2 hub.  By returning null, the jenkins plugin will fall back to trying to use legacy authentication, which won't work and the consequent errors will be misleading.